tests/ssg_test_suite/rule.py

```diff
@@ -110,14 +113,12 @@ def _apply_script(rule_dir, domain_ip, script):
     with open(log_file_name, 'a') as log_file:
         log_file.write('##### {0} / {1} #####\n'.format(rule_name, script))
 
-        command = ("ssh", machine, "cd {0}; bash -x {1}".format(rule_dir, script))
-        try:
-            subprocess.check_call(command,
-                                  stdout=log_file,
-                                  stderr=subprocess.STDOUT)
-        except subprocess.CalledProcessError as e:
-            logging.error("Rule testing script {0} failed with exit code {1}"
-                          .format(script, e.returncode))
+        command = "cd {0}; bash -x {1}".format(rule_dir, script)
+        args = common.IGNORE_KNOWN_HOSTS_OPTIONS + (machine, command)
+
+        rc = _run_with_stdout_logging("ssh", args, log_file)
+        if rc != 0:
```

Comment:
This way reminds me programming in C. However, in Python we use exception to handle problems.